stl/inc/xutility

```diff
@@ -2110,12 +2110,13 @@ namespace ranges {
 
             template <class _Ty>
             _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+                _STL_INTERNAL_CHECK(is_lvalue_reference_v<_Ty>);
                 if constexpr (is_array_v<remove_reference_t<_Ty>>) {
                     return {_St::_Array, true};

```

Comment:
I'm now thinking we should `static_assert` completeness of the element type here. A hard error (instead of "whatever behavior falls out of the code as written") better defends the "ill-formed with no diagnostic required" case and allows that behavior to be replaced more easily in the future if/when WG21 defines behavior for arrays of incomplete element type. (Applies to `end`, `rbegin`, and `rend` as well.)